les/client_handler.go

```diff
@@ -353,6 +398,25 @@ func (h *clientHandler) handleMsg(p *peer) error {
 		p.fcServer.ReceivedReply(resp.ReqID, resp.BV)
 		p.Log().Trace("Setting peer etherbase", "etherbase", resp.Etherbase, "Peer ID", p.ID)
 		p.SetEtherbase(resp.Etherbase)
+
+	case GatewayFeeMsg:
+		var resp struct {
+			ReqID, BV uint64
+			Data      GatewayFeeResps
+		}
+
+		if err := msg.Decode(&resp); err != nil {
+			return errResp(ErrDecode, "msg %v: %v", msg, err)
+		}
+
+		p.fcServer.ReceivedReply(resp.ReqID, resp.BV)
+
+		currGatewayFee := big.NewInt(int64(resp.Data.GatewayFee))
```

Comment:
Always handle gateway fee as a bit.Int or an appropriate wire format (e.g. hexutil.Big) casting to and
 from `int64` is unsafe.